server/src/test/java/org/elasticsearch/index/mapper/MappingLookupTests.java

```diff
@@ -53,20 +63,135 @@ public void testSubfieldOverride() {
         MockFieldMapper fieldMapper = new MockFieldMapper("object.subfield");
         ObjectMapper objectMapper = new ObjectMapper("object", "object", new Explicit<>(true, true), ObjectMapper.Nested.NO,
             ObjectMapper.Dynamic.TRUE, Collections.singletonMap("object.subfield", fieldMapper), Version.CURRENT);
-        MappingLookup mappingLookup = new MappingLookup(Collections.singletonList(fieldMapper), Collections.singletonList(objectMapper),
-            Collections.emptyList(), Collections.singletonList(new TestRuntimeField("object.subfield", "type")), 0);
+        MappingLookup mappingLookup = new MappingLookup(
+            "_doc",
+            Collections.singletonList(fieldMapper),
+            Collections.singletonList(objectMapper),
+            Collections.emptyList(),
+            Collections.singletonList(new TestRuntimeField("object.subfield", "type")),
+            0,
+            null,
+            false
+        );
         assertThat(mappingLookup.getMapper("object.subfield"), instanceOf(MockFieldMapper.class));
         assertEquals(1, size(mappingLookup.fieldMappers()));
         assertEquals(1, mappingLookup.objectMappers().size());
         assertThat(mappingLookup.fieldTypes().get("object.subfield"), instanceOf(TestRuntimeField.class));
         assertEquals(1, size(mappingLookup.fieldTypes().filter(ft -> true)));
     }
 
+
+    public void testAnalyzers() throws IOException {
+        FakeFieldType fieldType1 = new FakeFieldType("field1");
+        FieldMapper fieldMapper1 = new FakeFieldMapper(fieldType1, "index1");
+
+        FakeFieldType fieldType2 = new FakeFieldType("field2");
+        FieldMapper fieldMapper2 = new FakeFieldMapper(fieldType2, "index2");
+
+        MappingLookup mappingLookup = new MappingLookup(
+            "_doc",
+            Arrays.asList(fieldMapper1, fieldMapper2),
+            Collections.emptyList(),
+            Collections.emptyList(),
+            Collections.emptyList(),
+            0,
+            null,
+            false
+        );
+
+        assertAnalyzes(mappingLookup.indexAnalyzer("field1", f -> null), "field1", "index1");
+        assertAnalyzes(mappingLookup.indexAnalyzer("field2", f -> null), "field2", "index2");
+        expectThrows(IllegalArgumentException.class,
+            () -> mappingLookup.indexAnalyzer("field3", f -> {
+                throw new IllegalArgumentException();
+            }).tokenStream("field3", "blah"));
+    }
+
+    private void assertAnalyzes(Analyzer analyzer, String field, String output) throws IOException {
```

Comment:
s/Analyzes/Analyzers

Reply:
I think it is right as is. It is asserting that the term analyzes to the output term.